pkg/ebpf/tracer.go

```diff
@@ -249,7 +249,15 @@ func (t *Tracer) initPerfPolling() (*bpflib.PerfMap, error) {
 //  • Local DNS (*:53) requests if configured (default: true)
 func (t *Tracer) shouldSkipConnection(conn *ConnectionStats) bool {
 	isDNSConnection := conn.DPort == 53 || conn.SPort == 53
-	return !t.config.CollectLocalDNS && isDNSConnection && conn.Direction == LOCAL
+	switch {
+	case !t.config.CollectLocalDNS && isDNSConnection && conn.Direction == LOCAL:
+		return true
+	case conn.Direction == LOCAL || INCOMING:
```

Comment:
Not sure what connection direction has to do here. I'd imagine this is either the above condition or else

`util2.IsBlacklistedConnection("source", conn.Source, conn.SPort) || util2.IsBlacklistedConnection("destination", conn.Destination, conn.DPort)`